### PR TITLE
643 potential removal of dependencies

### LINKED ISSUE #643: Pre-release activities - Potential removal of dependencies
Part of #624 related to vignettes. Please make a PR to `pre-release@main`

Consider:

- [x] Removal of `{magrittr}` by importing pipe operator from `{dplyr}`   :heavy_check_mark: _done_
- [x] Other packages that can be easily removed   :heavy_check_mark: removed `tidyselect` as its functions are reexported from `dplyr`
- Move any of the packages from `Depends` -> `Imports`
  - [x] `{ggmosaic}`   :x: _impossible_ (see below)
  - [x] `{ggplot2}`   :x: _impossible_ (see below)
  - [x] `{shinyTree}`   :heavy_check_mark: _done_
  - [x] `{teal.transform}`   :x: _valid_: `teal.transform` functions should be available to `tmg` users
  - [x] `{teal}`   :x: _valid_: `teal` dependency is standard in all module packages
  - [x] `{shiny}`   :x: _valid_: like `teal`, module packages use `shiny` extensively

### PR DESCRIPTION
**Pull request by chlebowa**
Closes #643

### CODE REVIEW

**Comment by m7pr** [comment]
Just to summarize main changes
- `magrittr::%>%` is substittuted with `dplyr::%>%` hence `magrittr` is removed from Imports
- `teal.data` is upgraded from `Suggests` to `Imports` as `teal.data::datanames()` is widely used (e.g. `srv_front_page`)
- `shinyTree` is downgraded from `Depends` to `Imports` as it is used in `ui_viewer` of `tm_file_viewer`
- `methods` is removed from `Suggests`
- `library(nestcolor)` is changed to `require(nestcolor)` in examples so that `nestcolor` can remain in `Suggests` and we do not loose all examples (if we apply `@examplesIf requireNamespace` and `nestcolor` is not installed in examples)

**Comment by m7pr** [comment]
Will take a look on my own now.

**Comment by m7pr** [comment]
- `MASS::fitdistr` is used in `srv_distribution` of `tm_g_distribution` but it's only in `Suggests` - should we upgrade to `Imports`?
- `colourpicker::colourInput` is used in `tm_g_scatterplot` but it's only in `Suggests` - should we upgrade to `Imports`?
- `tools::file_ext` is used in `tm_file_viewer` but not in neither `Suggests`, `Imports` nor `Depends` - should we add to `Imports`?

All 3 packages would be caught if we did not use prefixes but would go with the other school that uses `@imporfFrom` manually for every foreign function in the package : D

**Comment by m7pr** [comment]
I actually used your code @chlebowa from here https://github.com/insightsengineering/idr-tasks/issues/632#issuecomment-1708247088 to detect `MASS`, `colourpicker` and `tools` not being added to `Imports`.

**Comment by chlebowa** [comment]
`tools` is an oversight and should be added to `Imports`, thank you. The other ones are covered by `if(requireNamespace)`, if I am not mistaken.

**Comment by chlebowa** [comment]
Funny that checks had passed without `Imports: tools`. Perhaps because `tools` is always installed as part of `r-base`?

**Comment by m7pr** [comment]
> tools is an oversight and should be added to Imports, thank you. The other ones are covered by if(requireNamespace), if I am not mistaken.

Sorry that I missed `colourpicker` and `MASS` : D they are used in `requireNamespace` with a `Filter` and vector input lol

```r
  extra_packages <- c("ggpmisc", "ggExtra", "colourpicker")
  missing_packages <- Filter(function(x) !requireNamespace(x, quietly = TRUE), extra_packages)
```

**Comment by m7pr** [comment]
> Funny that checks had passed without Imports: tools. Perhaps because tools is always installed as part of r-base?

Maybe, but R sometimes warns about `stats` or `grDevice` or `methods` which are a part of R Base, so not sure tbh.

**Comment by m7pr** [comment]
Ok, so besides tools, this is good to go. R CMD does not warn about it so I think we can skip it,

**Review by m7pr — APPROVED** [review]

File: DESCRIPTION [hunk 1]
```diff
@@ -36,21 +36,21 @@ Imports:
     forcats (>= 1.0.0),
     grid,
     logger (>= 0.2.0),
-    magrittr (>= 1.5),
     scales,
     shinyjs,
+    shinyTree,
```
[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION:41]: continued in here https://github.com/insightsengineering/teal.modules.general/pull/639#discussion_r1508764402

**Pull request merged**